ts/components/screens/BaseScreenComponent.tsx

```diff
@@ -21,24 +21,67 @@ import {
   openInstabugQuestionReport,
   openInstabugReplies,
   setInstabugSupportTokenAttribute,
+  setInstabugUserAttribute,
   TypeLogs
 } from "../../boot/configureInstabug";
 import I18n from "../../i18n";
 import customVariables from "../../theme/variables";
 import { setStatusBarColorAndBackground } from "../../utils/statusBar";
 import { handleItemOnPress } from "../../utils/url";
-import ContextualHelpModal from "../ContextualHelpModal";
+import ContextualHelpModal, {
+  RequestAssistancePayload
+} from "../ContextualHelp/ContextualHelpModal";
 import { SearchType } from "../search/SearchButton";
 import Markdown from "../ui/Markdown";
 import {
   deriveCustomHandledLink,
   isIoInternalLink
 } from "../ui/Markdown/handlers/link";
-import { SupportTokenState } from "../../store/reducers/authentication";
 import { getValueOrElse } from "../../features/bonus/bpd/model/RemoteValue";
-import { SupportToken } from "../../../definitions/backend/SupportToken";
 import { AccessibilityEvents, BaseHeader } from "./BaseHeader";
 
+/**
+ * Run side-effects from the Instabug library based on the type of support.
+ */
+function handleOnContextualHelpDismissed(
+  payload: RequestAssistancePayload,
+  attachmentConfig: DefaultReportAttachmentTypeConfiguration
+): void {
+  const maybeSupportToken = getValueOrElse(payload.supportToken, undefined);
+  const { supportType } = payload;
+
+  switch (supportType) {
+    case BugReporting.reportType.bug: {
+      // Store/remove and log the support token only if is a new assistance request.
+      // log on instabug the support token
+      if (maybeSupportToken) {
+        instabugLog(
+          JSON.stringify(maybeSupportToken),
+          TypeLogs.INFO,
+          "support-token"
+        );
+      }
+      // set or remove the support token as instabug user attribute
+      setInstabugSupportTokenAttribute(maybeSupportToken);
+
+      if (payload.deviceUniqueId) {
+        setInstabugUserAttribute("deviceUniqueID", payload.deviceUniqueId);
```

Comment:
I guess (actually I'm pretty sure) if a user sent its device ID, it persists also on future assistance requests
Could we remove this attribute if the user doesn't want to share it?
something like done in `setInstabugSupportTokenAttribute`

Reply:
Good catch.
I've changed [this line](https://github.com/pagopa/io-app/pull/3266/commits/e4e9abf83b7eb1efa3b1a75a66ebf4e12c6fe423#diff-99b6f65a74838b4555bfed0547c21e3f554b6fbca42f36eb9687795a8c2e96b6R67) and added an ad-hoc [test](https://github.com/pagopa/io-app/pull/3266/commits/e4e9abf83b7eb1efa3b1a75a66ebf4e12c6fe423#diff-88d9976dbc65c76157a0ed848973de00891123ff8787772003de32a36ff0d508R101).

While doing it, I've polished a bit [the Instabug helpers](https://github.com/pagopa/io-app/pull/3266/commits/e4e9abf83b7eb1efa3b1a75a66ebf4e12c6fe423#diff-e8e418fa64073850d308f5d207dcdde4fdc42dfe50a21276c96b8cce3c12a4c1R142) and added [missing tests](ts/boot/__tests__/configureInstabug.test.ts).